### ISSUE TITLE
WriteFeaturesMarkFDK.py: template glyphs cause an error

### ISSUE THREAD

**Issue by rbmntjs** [adobe-type-tools/python-modules#6]
I think it would make sense to allow for empty template glyphs – right now, a `.ufo` with some unfilled slots will fail the mark writer.

**Comment by miguelsousa** [comment]
What do you mean by *empty template glyphs*? Are these glyphs actually present in the UFO font?

**Comment by miguelsousa** [comment]
Also, if you're getting a traceback please provide it.

**Comment by rbmntjs** [comment]
In RoboFont, this is a grey glyph box: part of the active charset but not actually filled in. I only know it as a template glyph. Not sure what else to call it.

<img width="188" alt="screen shot 2016-06-15 at 23 27 13" src="https://cloud.githubusercontent.com/assets/49890/16097896/b77bbc40-3350-11e6-8c20-ca3e448c896f.png">

Traceback is the following.
```
Traceback (most recent call last):
  File "marker.py", line 15, in <module>
    WriteFeaturesMarkFDK.MarkDataClass(font, os.path.dirname(font.path), trimCasingTags, genMkmkFeature, writeClassesFile, indianScriptsFormat)
  File "/Users/rob/Dropbox/Type/30_Code/tools/python-modules/WriteFeaturesMarkFDK.py", line 237, in __init__
    self.collectAnchorDataFromBaseGlyphs()
  File "/Users/rob/Dropbox/Type/30_Code/tools/python-modules/WriteFeaturesMarkFDK.py", line 369, in collectAnchorDataFromBaseGlyphs
    glyphs.append(self.f[name])
  File "/Library/Python/2.7/site-packages/RoboFab/Lib/robofab/objects/objectsBase.py", line 699, in __getitem__
    return self.getGlyph(glyphName)
  File "/Library/Python/2.7/site-packages/RoboFab/Lib/robofab/objects/objectsRF.py", line 558, in getGlyph
    raise KeyError, glyphName
KeyError: 'prescription'
```

**Comment by miguelsousa** [comment]
Got it. The term template glyph is correct. You didn't mention RoboFont initially, so I wasn't sure. The tool can be used outside of RF.

I haven't looked into it yet, but this should be easy to fix. I believe that the names of the template glyphs are listed on the UFO's `public.glyphOrder`, but the glyphs don't actually exist in the font. The script needs to skip those names.

**Comment by frankrolf** [comment]
Robofont has the method `f.glyphOrder`, which returns only filled-in glyphs. 
`f.lib['public.glyphOrder']` may also contains template glyphs (but is necessary for running the script from the command line).
Since the environment is tested at the beginning of the script, you may just add something like:
```
if self.appName == 'Robofont':
    glyphOrder = f.glyphOrder
else:
    glyphOrder = self.f.lib["public.glyphOrder"] 
```

**Comment by miguelsousa** [comment]
@frankrolf that would fix things for RoboFont, but it would still be broken when running from the command line.

**Comment by miguelsousa** [comment]
Fixed by e155395c88dea9253509b67aea39a9b2d6bc8ce4
